sycl/test-e2e/Matrix/joint_matrix_apply_two_matrices_impl.hpp

```diff
@@ -8,19 +8,24 @@
 #include <sycl/usm.hpp>
 
 template <typename Tc, typename Ta, size_t M, size_t N>
-bool apply_verify(Tc *C, Tc *D, Ta *A, Ta *Ar) {
+bool apply_verify(Tc *C, Tc *D, Ta *A, Ta *Ar, Tc *Cref, Ta *Aref) {
```

Comment:
I don't like that we duplicate matrix_compare functionality, and it may cause more bugs, like below - comparison to float epsilon was added, but isnan check was not added for D and Ar.
I suggest to redesign code of apply_verify to start using matrix_compare for matrix comparisons.
Maybe something like this:
Introduce new function to common.hpp. Something like:
`matrix_apply(mat1, mat2, op)`
Then we can modify `apply_verify` here to be like:
```
allocate and make a copy matref1 to matref2, since we need 2 different matrices to compare to. then:
matrix_apply(matref1, matref2, add2);  // same operation that is happening on target
return matrix_compare(matref1, mattarget1) && matrix_compare(matref2, mattarget2);
```
Then in the place where verify is called it can be something like:
`return apply_verify<>(C, D, Cref) && apply_verify<>(A, Ar, Aref)

Reply:
I am fine with the change but it sounds like we may end up with more intermediate matrices in storage.